src/reducers/configuration.js

```diff
@@ -0,0 +1,40 @@
+"use strict";
+
+var validateConfiguration = require("../validate/validate-configuration");
+var isFunction = require("../utils/is-function");
+
+var initialState = {
+  httpAddresses: [],
+  wsAddresses: [],
+  ipcAddresses: [],
+  connectionTimeout: 3000,
+  pollingIntervalMilliseconds: 30000,
+  blockRetention: 100,
+  errorHandler: null
+};
+
+module.exports = function (configuration, action) {
+  var updatedConfiguration;
+  if (typeof configuration === "undefined") {
+    return initialState;
+  }
+  switch (action.type) {
+    case "SET_CONFIGURATION":
```

Comment:
Users aren't expected to call the `SET_CONFIGURATION` action itself, though.  The raw actions are technically accessible, but they're not part of the normal public API.

I think storing configuration in Redux makes sense (minus the out-of-band error handler, which is now stored elsewhere).  It's simple, serializable, and quite relevant to ethrpc's behavior -- if you were going to rehydrate and/or rewind, it seems to me you'd lose an awful lot of context if you didn't know (for example) what endpoints were configured at the time.